### PR TITLE
Fix possible crash in report copy progress (#464)

### LINKED ISSUE #464: Crash when trying to duplicate folder via copy-pasting
As in title. Attempt to Ctrl-C Ctrl-V folder results in crashing
```
[FATAL 21:05:31.399366] [GLib-GIO] g_file_query_info: assertion 'G_IS_FILE (file)' failed
[FATAL 21:05:31.399394] [GLib-GIO] g_file_get_basename: assertion 'G_IS_FILE (file)' failed
```
followed by segfault

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/58114184-crash-when-trying-to-duplicate-folder-via-copy-pasting?utm_campaign=plugin&utm_content=tracker%2F65602118&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F65602118&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes #464  (hopefully)

Although this crash could not be reproduced on the author's machine, the backtrace indicated message printing would crash if copy job data became invalid due to a race or otherwise.

This PR checks that the source and destination file pointers in the copy job are still valid before reporting progress.  Also prevents some custom print handlers crashing if passed an invalid file pointer.

### CODE REVIEW

**Comment by sakralbar** [comment]
If you need help for testing with this (or [this](https://github.com/elementary/files/issues/464)) problem, please tell me.

**Comment by jeremypw** [comment]
@sakralbar : Hi!  Thanks for the offer. If you could build and test this branch and provide a review then, if you approve the PR, I can ask for it to be merged by another team member on that basis.

**Comment by sakralbar** [comment]
Sorry, I tried to build this package, but I got an error:

```
eugene@eugene-VirtualBox:~/files/build$ cmake -DCMAKE_INSTALL_PREFIX=/usr ..
CMake Error: The source directory "/home/eugene/files" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
```
And I could not find this file.

**Comment by jeremypw** [comment]
@sakralbar :  The latest version of Files now only builds using meson I am afraid.  The cmake files have only recently been removed. The README.md needs updating! 

Install the `meson` package from the repositories using synaptic or apt (its not in appcenter).

You can build by running  `meson build --prefix=/usr && cd build && ninja` from the root source directory.

You can then install with `sudo ninja install`

Sorry I forgot to mention that.

**Comment by sakralbar** [comment]
Ok, now it works.
It's screen recording [before](https://yadi.sk/i/8zeLEnp4sKReXA) and [after](https://yadi.sk/i/1f1II2gcS_bWig) building Files from source.

**Comment by jeremypw** [comment]
@sakralbar Thanks taking the trouble to test this and confirm it fixes the problem in your case :-)

**Comment by sakralbar** [comment]
@jeremypw Thank you too. I hope that soon there will be an update of Files in the App center, and the Elementary OS will have more Russian-speaking users.

File: libcore/marlin-file-operations.c [hunk 1]
```diff
@@ -802,6 +807,18 @@ custom_basename_to_string (char *format, va_list va)
     return name;
 }
 
+static char *
+custom_basename_to_string (char *format, va_list va)
+{
+    GFile *file;
+    GFileInfo *info;
+    char *name, *basename, *tmp;
```
Comment (donadigo) [inline libcore/marlin-file-operations.c]: I suppose `info`, `name`, `basename`, and `tmp` are not needed in this function?

**Comment by jeremypw** [comment]
@donadigo :  Is this OK now?

**Pull request merged**